gnovm/stdlibs/stdlibs.go

```diff
@@ -262,6 +263,59 @@ func InjectPackage(store gno.Store, pn *gno.PackageNode) {
 				m.PushValue(res0)
 			},
 		)
+		pn.DefineNative("PrevRealm",
+			gno.Flds( // params
+			),
+			gno.Flds( // results
+				"", "Realm",
+			),
+			func(m *gno.Machine) {
+				var (
+					ctx = m.Context.(ExecContext)
+					// Default lastCaller is OrigCaller, the signer of the tx
+					lastCaller  = ctx.OrigCaller
+					lastPkgPath = ""
+				)
+
+				for i := m.NumFrames() - 1; i > 0; i-- {
+					fr := m.Frames[i]
+					if fr.LastPackage == nil || !fr.LastPackage.IsRealm() {
+						// Ignore non-realm frame
+						continue
+					}
+					pkgPath := fr.LastPackage.PkgPath
+					// The first realm we encounter will be the one calling
+					// this function; to get the calling realm determine the first frame
+					// where fr.LastPackage changes.
+					if lastPkgPath != "" && lastPkgPath != pkgPath {
+						lastCaller = fr.LastPackage.GetPkgAddr().Bech32()
+						lastPkgPath = pkgPath
+						break
+					}
+
+					lastPkgPath = pkgPath
+				}
+
+				// Empty the pkgPath if we return a user
+				if ctx.OrigCaller == lastCaller {
+					lastPkgPath = ""
```

Comment:
`std.Realm` return a Realm{}

```go
type Realm struct {
	Addr    crypto.Bech32Address
	PkgPath string
}
```

in the case we return a user, we want PkgPath to be empty.

Without this line, we could return a Realm containing a user address and `gno.land/r/demo/xxx` which is the "Current realm".

This condition won't be present in the next function `std.Realm`

Reply:
For accounts, it's a good idea to have an empty variable. However, if default realm implementation is desired later, it could change to `gno.land/r/moul`. What is the current value before erasing it?

I like using `if PkgPath != ""` for distinction, similar to `if err != nil`. We can update the method or choose a new variable when accounts have a path.

Waiting for your response, but currently inclined to maintain the current behavior.